march_rqt_input_device/src/march_rqt_input_device/input_device_view.py

```diff
@@ -0,0 +1,301 @@
+import os
+
+
+from python_qt_binding import loadUi
+from python_qt_binding.QtCore import QSize
+from python_qt_binding.QtWidgets import QGridLayout
+from python_qt_binding.QtWidgets import QPushButton
+from python_qt_binding.QtWidgets import QWidget
+from qt_gui.plugin import Plugin
+import rospkg
+
+from march_rqt_input_device.input_device_controller import InputDeviceController
+
+
+class InputDevicePlugin(Plugin):
+    def __init__(self, context):
+        super(InputDevicePlugin, self).__init__(context)
+        self.controller = InputDeviceController()
+
+        # region initialising the widget
+        self.setObjectName('InputDevicePlugin')
+
+        # Create QWidget
+        self._widget = QWidget()
+
+        # Get path to UI file which should be in the 'resource' folder of this package
+        ui_file = os.path.join(rospkg.RosPack().get_path('march_rqt_input_device'), 'resource', 'input_device.ui')
+
+        # Extend the widget with all attributes and children from UI file
+        loadUi(ui_file, self._widget)
+
+        # Give QObjects reasonable names
+        self._widget.setObjectName('Input Device')
+
+        # Show _widget.windowTitle on left-top of each plugin (when it's set in _widget). (useful for multiple windows)
+        if context.serial_number() > 1:
+            self._widget.setWindowTitle('{0} ({1})'.format(self._widget.windowTitle(), context.serial_number()))
+
+        # Add widget to the user interface
+        context.add_widget(self._widget)
+
+        # Create buttons here
+        rocker_switch_increment = \
+            self.create_button('rocker_switch_up', color_code='#239b56',
+                               callback=lambda: self.controller.publish_rocker_switch(1))
+
+        rocker_switch_decrement = \
+            self.create_button('rocker_switch_down', color_code='#239b56',
+                               callback=lambda: self.controller.publish_rocker_switch(-1))
+
+        home_sit = \
+            self.create_button('home_sit', image_path='/home_sit.png',
+                               callback=lambda: self.controller.publish_gait('home_sit'))
```

Comment:
You could create a `create_gait_button` method that automatically creates a button from a given gait name with image and publish callback, since those are just the name.

Reply:
Creating such a function besides the current create_button function would not save much code and the consistency in usage of the current more generic function to me seems better,